CHANGES.md

```diff
@@ -1,8 +1,461 @@
+* improved formatting of formal arguments in generated code for --gen-ids
+
+Third-Party Software Changes
+----------------------------
+* updated to GASNet 1.28.0
+* updated to qthreads 1.12
+* updated to jemalloc 4.5.0
+* updated hwloc to 1.11.6
+* made GASNet build 'amudprun' using CHPL_HOST_COMPILER rather than ..TARGET..
+* limited the maximum size of qthreads memory pools to 65MB
+* made 'hwloc' always use libnuma for non-flat locale models when available
+* made libunwind's optional LZMA dependency work better (TODO: ???)
+
+Testing System
+--------------
+* added --multi-locale-only and --max-locales flags to 'start_test'
+* added a '-dirs' option to paratest.server to specify directories to test
+* added an '-env' flag to paratest.server for forwarding environment variables
+* added the ability to filter on graph names in the performance tracker
+* made the test system work even when CHPL_UNWIND is set
+* added a checker for the ANNOTATIONS.yaml file used by performance tracking
+* added overlays of the nightly performance graphs to the release-over-release
+
+Removed Features
+----------------
+* removed deprecated functions from the 'Sort' and 'Search' modules
+* removed the deprecated 'BigInt' class in favor of the 'bigint' value type
+* removed the 'noRefCount' config const which is no longer necessary
+
+Developer-oriented changes: Module changes
+------------------------------------------
+* refactored locale models code
+* optimized out array fields used in numa configurations only using 'void'
+* added 'export' to locale model routines to avoid widening arguments
+* refactored rectangular I/O into a single helper routine
+* moved c_void_ptr comparisons from CPtr to ChapelBase
+* added a developer feature for specifying arrays' initialization approaches
+* removed an unnecessary argument from _bulkGrow
+* simplified the 'dsi' interface for slicing, rank-change, and reindexing
+  (see TODO)
+* updated the 'dsi' interfaces for establishing new domains
+* cleaned up how the MPI module deinitializes/finalizes itself
+
+Developer-oriented changes: Makefile improvements
+-------------------------------------------------
+* improved the robustness of 'make check'
+* worked around a race condition in GASNet's Makefiles
+* cleaned up the runtime include paths
+* improved linking of libnuma when using the memkind library
+* improved Makefile portability for Bourne and limited shells
+
+Developer-oriented changes: Compiler Flags
+------------------------------------------
+
+Developer-oriented changes: Compiler improvements/changes
+---------------------------------------------------------
+* made PRIM_GETCID return an int(32) rather than a bool
+* replaced four FnSymbol fields and a flag with a std::map
+* improved the compiler's opt-in verification pass
+* udpated the compiler's representation of 'ref' types to use qualified types
+* fiexd a segfault in list_ast() for type-less symbols
```

Comment:
I can run spell-check on the file if you want to focus on whether your changes were captured.